include/pybind11/pybind11.h

```diff
@@ -2026,6 +2042,19 @@ void register_exception_translator(ExceptionTranslator&& translator) {
         std::forward<ExceptionTranslator>(translator));
 }
 
+
+/**
+  * Add a new module-local exception translator. Locally registered functions
+  * will be tried before any globally registered exception translators, which
+  * will only be invoked if the moduke-local handlers do not deal with
+  * the exception.
+  */
+template <typename ExceptionTranslator>
+void register_local_exception_translator(ExceptionTranslator&& translator) {
```

Comment:
Okay, I did this. I did not change any of the type information, because I'm pretty sure that would PYBIND11_INTERNALS_VERSION ABI breaking change (so internals still has
```
std::forward_list<ExceptionTranslator> registered_exception_translators;
```
not
```
std::forward_list<const ExceptionTranslator> registered_exception_translators;
```
as it would in your PR.

But I removed all the templating from the register functions. I also turned `ExceptionTranslator` into a `using` alias, which makes the typing a little bit nicer to read.

Reply:
Yes, perfect, thanks. I did not mean to suggest adding the const. My PR was just to record that I thought about it carefully and then discarded the idea because I think it's too disruptive for too little gain.